packages/yarnpkg-core/sources/Project.ts

```diff
@@ -1679,7 +1679,12 @@ export class Project {
         report.reportError(MessageName.IMMUTABLE_CACHE, `${this.configuration.format(ppath.basename(entryPath), `magenta`)} appears to be unused and would marked for deletion, but the cache is immutable`);
       } else {
         report.reportInfo(MessageName.UNUSED_CACHE_ENTRY, `${this.configuration.format(ppath.basename(entryPath), `magenta`)} appears to be unused - removing`);
-        await xfs.unlinkPromise(entryPath);
```

Comment:
You can use `xfs.removePromise` which handles both

Reply:
Ace